### PR TITLE
Fix reading SAS7BDAT files with zero rows

### LINKED ISSUE #18198: Reading a SAS file with 0 rows gives None
#### Code Sample, a copy-pastable example if possible

```python
In [2]: import pandas as pd

In [3]: pd.read_csv("zero_observations.csv")
Out[3]: 
Empty DataFrame
Columns: [a, b, c, d]
Index: []

In [4]: pd.read_sas("zero_observations.sas7bdat")

In [5]: 
```

This is `a.csv`:
```
$ cat zero_observations.csv
a,b,c,d
```

#### Problem description

When reading a SAS file with 0 records, it gives `None`. But when reading a CSV file with 0 records it gives an empty dataframe.

In SAS, we can atleast identify the datatypes correctly and make an empty dataframe with this.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.21.0
pytest: 3.2.1
pip: 9.0.1
setuptools: 23.0.0
Cython: 0.27.3
numpy: 1.13.3
scipy: 0.18.1
pyarrow: None
xarray: None
IPython: 6.1.0
sphinx: 1.4.9
patsy: None
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.0
openpyxl: None
xlrd: 1.1.0
xlwt: None
xlsxwriter: 0.9.8
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8.1
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by jonashaag**
- [x] closes #18198 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/io/sas/test_sas7bdat.py [hunk 1]
```diff
@@ -216,6 +216,15 @@ def test_zero_variables(datapath):
         pd.read_sas(fname)
 
 
+def test_zero_rows(datapath):
+    # GH 18198
+    fname = datapath("io", "sas", "data", "zero_rows.sas7bdat")
+    df = pd.read_sas(fname)
+    tm.assert_frame_equal(
+        df, pd.DataFrame([{"char_field": "a", "num_field": 1.0}]).iloc[:0]
```
Comment (mroeschke) [inline pandas/tests/io/sas/test_sas7bdat.py:224]: Nit: Could you set these to `result` and `expected` and then call `tm.assert_frame_equal(result, expected)`?

**Comment by jreback** [comment]
cc @bashtage  if you can look

**Comment by bashtage** [comment]
Looks correct and good to me.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jonashaag

**Pull request merged**